controllers/eventhandlers/service.go

```diff
@@ -2,169 +2,78 @@ package eventhandlers
 
 import (
 	"context"
-
-	corev1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/types"
-
+	"github.com/aws/aws-application-networking-k8s/pkg/apis/applicationnetworking/v1alpha1"
+	"github.com/aws/aws-application-networking-k8s/pkg/k8s"
 	"github.com/aws/aws-application-networking-k8s/pkg/model/core"
 	"github.com/aws/aws-application-networking-k8s/pkg/utils/gwlog"
-
+	corev1 "k8s.io/api/core/v1"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/handler"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
-
-	"k8s.io/client-go/util/workqueue"
-	"sigs.k8s.io/controller-runtime/pkg/event"
-	mcs_api "sigs.k8s.io/mcs-api/pkg/apis/v1alpha1"
 )
 
-type enqueueRequestForServiceWithExportEvent struct {
+type serviceEventHandler struct {
 	log    gwlog.Logger
 	client client.Client
+	mapper *resourceMapper
 }
 
-func NewEqueueRequestServiceWithExportEvent(log gwlog.Logger, client client.Client) handler.EventHandler {
-	return &enqueueRequestForServiceWithExportEvent{
-		log:    log,
-		client: client,
-	}
-}
-
-func (h *enqueueRequestForServiceWithExportEvent) Create(e event.CreateEvent, queue workqueue.RateLimitingInterface) {
-	h.log.Info("Event: service create")
-	service := e.Object.(*corev1.Service)
-	h.enqueueImpactedService(queue, service)
-	h.enqueueImpactedServiceExport(queue, service)
-}
-
-func (h *enqueueRequestForServiceWithExportEvent) Update(e event.UpdateEvent, queue workqueue.RateLimitingInterface) {
-}
-
-func (h *enqueueRequestForServiceWithExportEvent) Delete(e event.DeleteEvent, queue workqueue.RateLimitingInterface) {
-	h.log.Info("Event: service delete")
-	service := e.Object.(*corev1.Service)
-	h.enqueueImpactedService(queue, service)
-	h.enqueueImpactedServiceExport(queue, service)
+func NewServiceEventHandler(log gwlog.Logger, client client.Client) *serviceEventHandler {
+	return &serviceEventHandler{log: log, client: client,
+		mapper: &resourceMapper{log: log, client: client}}
```

Comment:
should we create mapper in the main func and pass mapper into controllers?

Reply:
This is worth considering, but what would we gain from such dependency injection? I did this way because mapper struct is defined mostly for easier unit testing and code reuse, and I don't think we will be configuring anything on the mapper.